### PR TITLE
Enable debugging itxn programs

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Set special addressed in EvalParams when evaluating from tealdbg/dryrun

## Test Plan

Tests added

### CODE REVIEW

File: daemon/algod/api/server/v2/dryrun.go [hunk 1]
```diff
@@ -338,41 +338,6 @@ func (dl *dryrunLedger) GetCreatorForRound(rnd basics.Round, cidx basics.Creatab
 	return basics.Address{}, false, fmt.Errorf("unknown creatable type %d", ctype)
 }
 
-func (dl *dryrunLedger) getAppParams(addr basics.Address, aidx basics.AppIndex) (params basics.AppParams, err error) {
```
Comment (algorandskiy) [inline daemon/algod/api/server/v2/dryrun.go:341]: removed some unused code

File: daemon/algod/api/server/v2/dryrun_test.go [hunk 2]
```diff
@@ -1331,3 +1332,95 @@ func TestDryrunCost(t *testing.T) {
...
+			},
+		},
+	}
+
+	dr.ProtocolVersion = string(dryrunProtoVersion)
+
+	var response generated.DryrunResponse
+	doDryrunRequest(&dr, &response)
+	require.NoError(t, err)
+	checkAppCallPass(t, &response)
```
Comment (jasonpaulos) [inline daemon/algod/api/server/v2/dryrun_test.go:1422]: It seems like we neglected to add inner transactions to the `DryrunTxnResult` API type. @jannotti was there a reason for this, or did we just forgot about it?

Ideally any created inner transactions would be returned in that response and this test could verify they are there and properly formatted.

Comment (jannotti) [reply daemon/algod/api/server/v2/dryrun_test.go:1422]: Definitely not on purpose.

Comment (jannotti) [reply daemon/algod/api/server/v2/dryrun_test.go:1422]: I am incapable of expressing how much I hate that dryrun lives in its own universe where we have to do everything all over again.

**Review by jannotti — APPROVED** [review]

**Pull request merged**